### ISSUE TITLE
Hard to debug with exceptions captured

### ISSUE THREAD

**Issue by MikhailArkhipov** [Microsoft/RTVS#2320]
If exception is re-thrown from capture debugger always breaks (by design). It is very hard to get out of this and continue. 

![image](https://cloud.githubusercontent.com/assets/12820357/18495027/4e2a4fd4-79cf-11e6-98aa-a63e6a443096.png)

**Comment by AlexanderSher** [comment]
What is a desired behavior?

**Comment by MikhailArkhipov** [comment]
Not entirely sure. Debugger folks said it was by design and debugger breaks in this case even when checkbox is off... It feels like in the debugger F5 does not proceed as break happens endlessly.

**Comment by int19h** [comment]
To respect the "Break when this exception type is thrown" checkbox, I believe.

However, as I recall, that checkbox is mislabeled - it's actually "Break when this exception type is thrown _and handled_". So it doesn't apply to unhandled exceptions. In this case, the exception that you're observing is, indeed, unhandled - it's coming out of a task on which `DoNotWait` was applied, so it's not possible to handle, which is why it's trying to fail-fast the process by rethrowing that extension on a threadpool thread. That's why the dialog doesn't go away - the only thing that can be done here otherwise is to terminate the process.

Basically, the bug is whatever exception being rethrown here occurring in the first place. It should either be handled normally, or silenced with `SilenceException`, at the point where `DoNotWait` was called.

**Comment by MikhailArkhipov** [comment]
It is probably worth re-throwing critical exceptions only.

**Comment by int19h** [comment]
Why? If an exception is thrown in regular code, it doesn't matter if it's critical or not - you either handle it on some level, or it crashes the app. This forces exceptions to be at least inspected (if you want to suppress one, you have to do so explicitly).

The logic here is just translating that to async tasks - if it can throw, either observe it or explicitly silence it. Silently swallowing stuff just leads to masking bugs. In this case, for example, it means that we're not handling that 401 error coming from the server, and presumably not providing proper indication of error to the user.

**Comment by MikhailArkhipov** [comment]
I guess I'll rephrase. The exception is important but may not be fixable or relevant at the moment. In order to unblock debugging one may need some means to control what is being rethrown so when they do get the exception, they can file a bug on the code owner and meanwhile continue on their debugging tasks at hand without having to resort to Debug | Terminate and starting over.

**Comment by int19h** [comment]
This is an exception that would crash the application if the debugger was not attached, so I don't think there's any way to treat it as something that can be meaningfully worked around. This is also same as regular, non-async exception handling - if there's a `throw` without a matching catch, you'll keep getting the exception dialog when you F5 again and again, because there's nowhere to continue to - all you can do is Terminate.
